sql/mozfun/glam/fenix_nightly_geckoview_versions/stored_procedure.sql

```diff
@@ -0,0 +1,171 @@
+/*
+Enumerate the most common GeckoView version for each build hour for "nightly"
+Fenix builds.
+
+This includes all hours up to the reference day e.g. the last row up where the
+reference date is 2020-10-01 will include hours up to midnight of that day.
+
+Example:
+
+```
+DECLARE results ARRAY<
+    STRUCT<build_hour DATETIME, geckoview_version STRING, n_builds INT64>
+>;
+CALL fenix_nightly_geckoview_versions(DATE "2020-10-01", 28, results);
+SELECT * FROM UNNEST(results)
+```
+
+The first row will be (2020-09-03T00:00:00, 82.0a1, 39901).
+The last row will be (2020-10-01T00:00:00, 83.0a1, 46716).
+*/
+CREATE OR REPLACE PROCEDURE
+  glam.fenix_nightly_geckoview_versions(
+    IN reference_date DATE,
+    IN history_days INT64,
+    OUT build_hour_version ARRAY<
+      STRUCT<build_hour DATETIME, geckoview_version STRING, n_builds INT64>
+    >
+  )
+BEGIN
+  -- See the fenix_build_to_datetime udf for details on Fenix build internals.
+  -- https://github.com/mozilla/bigquery-etl/blob/56defb1061445115277d90def06ad41185e519b5/sql/moz-fx-data-shared-prod/udf/fenix_build_to_datetime/metadata.yaml#L1-L23
+  SET build_hour_version = (
+    WITH extracted AS (
+    -- We'll look at the metrics ping to estimate the major geckoview version.
+    -- The metrics section is aliased, so we must rename the table for this to
+    -- work as expected.
+      SELECT
+        submission_timestamp,
+        client_info.app_build,
+        metrics.string.geckoview_version,
+      FROM
+        `moz-fx-data-shared-prod`.org_mozilla_fenix.metrics AS t1
+      WHERE
+        mozfun.norm.fenix_app_info('org_mozilla_fenix', client_info.app_build).channel = 'nightly'
+      UNION ALL
+      SELECT
+        submission_timestamp,
+        client_info.app_build,
+        metrics.string.geckoview_version,
+      FROM
+        `moz-fx-data-shared-prod`.org_mozilla_fenix_nightly.metrics AS t1
+      UNION ALL
+      SELECT
+        submission_timestamp,
+        client_info.app_build,
+        metrics.string.geckoview_version,
+      FROM
+        `moz-fx-data-shared-prod`.org_mozilla_fennec_aurora.metrics AS t1
+    ),
+    transformed AS (
+      SELECT
+        app_build,
+        geckoview_version,
+        -- Truncate to the hour, since older builds give minute resolution.
+        datetime_TRUNC(
+          `moz-fx-data-shared-prod`.udf.fenix_build_to_datetime(app_build),
+          HOUR
+        ) AS build_hour
+      FROM
+        extracted
+      WHERE
+        DATE(submission_timestamp) >= DATE_SUB(reference_date, INTERVAL history_days DAY)
```

Comment:
Should this be `BETWEEN DATE_SUB(reference_date, INTERVAL history_days DAY) AND reference_date`?

Reply:
Yes, that's a good catch.